tools/check-typo

```diff
@@ -127,9 +127,12 @@ IGNORE_DIRS="
     if $is_svn || $is_cmd_line; then :; else continue; fi
     svnrules=''
     if $is_svn; then
-      case `git check-attr binary "$f"` in
-        *'binary: set') continue;;
-      esac
+      # 4b825d is automatically in every repository (see cache.h)
+      # diff-tree is plumbing, so the output is stable and -\t- is the
+      # guaranteed output for a binary file only
+      EMPTY=4b825dc642cb6eb9a060e54bf8d69288fbee4904
+      git diff-tree --numstat $EMPTY HEAD -- "$f" | grep -q "^-[[:blank:]]-" \
+        && continue
```

Comment:
Comment expanded - in fact, your use of git hash-tree reads better in the code than a hard-coded SHA (and speed is hardly important here!). I've added direct links to the git documentation for diff-tree and a note that the arcane as it looks, it is supposed to be stable (because it uses git "plumbing" rather than "porcelain")!